backends/bmv2/psa_switch/psaSwitch.cpp

```diff
@@ -648,7 +667,36 @@ Util::IJson* ExternConverter_InternetChecksum::convertExternObject(
     UNUSED ConversionContext* ctxt, UNUSED const P4::ExternMethod* em,
     UNUSED const IR::MethodCallExpression* mc, UNUSED const IR::StatOrDecl *s,
     UNUSED const bool& emitExterns) {
-    auto primitive = mkPrimitive("InternetChecksum");
+    Util::JsonObject* primitive=nullptr;
+    if (mc->arguments->size()<2)
+        primitive = mkPrimitive("_" + em->originalExternType->name +
+                                 "_" + em->method->name);
+    else
+        primitive = mkPrimitive("_" + em->originalExternType->name +
+                                 "_" + "get_verify");
+    auto parameters = mkParameters(primitive);
+    primitive->emplace_non_null("source_info", s->sourceInfoJsonObj());
+    auto cksum = new Util::JsonObject();
+    cksum->emplace("type", "extern");
+    cksum->emplace("value", em->object->controlPlaneName());
+    parameters->append(cksum);
+    if (em->method->name == "add" || em->method->name == "subtract") {
+        auto fieldList=new Util::JsonObject();
+        fieldList->emplace("type","field_list");
+        auto fieldsJson = ctxt->conv->convert(mc->arguments->at(0)->expression, true, false);
+        fieldList->emplace("value",fieldsJson);
+        parameters->append(fieldList);
+    } else if (em->method->name != "clear") {
+        if (mc->arguments->size() == 2) {  // get_verify
```

Comment:
If people use the wrong model psa.p4 when compiling these assumptions may not be true. 
You should really check them and give a model error if they are not satisfied.

Reply:
I will add the number of arguments check.